jps-plugin/src/org/intellij/erlang/jps/builder/ErlangBuilder.java

```diff
@@ -266,31 +254,40 @@ private static void addIncludePaths(@NotNull GeneralCommandLine commandLine, @Nu
   @NotNull
   private static List<String> getErlangModulePaths(@NotNull ErlangTarget target,
                                                    @NotNull CompileContext context,
-                                                   @NotNull List<String> dirtyFilePaths,
                                                    boolean isTest) {
-    List<ErlangFileDescriptor> fileDescriptors = getErlangFileDescriptorFromConfig(target, context, isTest);
-    return fileDescriptors != null ?
-           getSortedErlangModulePathsToCompile(fileDescriptors, dirtyFilePaths) :
-           getErlangModulePathsDefault(target, isTest);
+    List<String> erlangBuilderDirtyFilesInfo = getErlangModulePathsFromConfig(target, context, isTest);
+    return erlangBuilderDirtyFilesInfo != null ? erlangBuilderDirtyFilesInfo : getErlangModulePathsDefault(target, isTest);
   }
 
-  @NotNull
-  private static List<String> getSortedErlangModulePathsToCompile(@NotNull List<ErlangFileDescriptor> sortedModuleDescriptors,
-                                                                  @NotNull List<String> dirtyModules) {
-    SortedModuleDependencyGraph semiGraph = new SortedModuleDependencyGraph(sortedModuleDescriptors);
-    GraphGenerator<Node> graph = GraphGenerator.create(semiGraph);
-    markDirtyNodes(semiGraph.getNodesByName(dirtyModules), graph);
-    return ContainerUtil.mapNotNull(semiGraph.getNodes(), new Function<Node, String>() {
-      @Nullable
-      @Override
-      public String fun(Node node) {
-        return node.myDirty && FileUtilRt.extensionEquals(node.myErlangModulePath, "erl") ? node.myErlangModulePath : null;
-      }
-    });
+  @Nullable
+  private static List<String> getErlangModulePathsFromConfig(@NotNull ErlangTarget target,
+                                                             @NotNull CompileContext context,
+                                                             boolean isTests) {
+    String fileName = ERLANG_BUILDER_DIRTY_FILES_PREFIX + target.getModule().getName() + ".xml";
+    ErlangModuleBuildOrder erlangModuleBuildOrder = null;
+    try {
+      erlangModuleBuildOrder = ErlangBuilderXmlUtil.readFromXML(fileName, context, ErlangModuleBuildOrder.class);
+    }
+    catch (JDOMException e) {
+      LOG.warn("Can't read XML from " + fileName);
+      LOG.debug(e);
+    }
+    catch (IOException e) {
+      LOG.warn("I/O exceptions is occurred while reading " + fileName);
+      LOG.debug(e);
```

Comment:
We'll know it's an IOException - log something meaningful, like "Can't read %filename%".

Log exceptions at warn level - we'll want them in logs if they occur.